ext/sysvmsg/sysvmsg.c

```diff
@@ -73,17 +72,58 @@ zend_module_entry sysvmsg_module_entry = {
 ZEND_GET_MODULE(sysvmsg)
 #endif
 
-static void sysvmsg_release(zend_resource *rsrc)
+/* SysvMessageQueue class */
+
+zend_class_entry *sysvmsg_queue_ce;
+static zend_object_handlers sysvmsg_queue_object_handlers;
+
+static inline sysvmsg_queue_t *sysvmsg_queue_from_obj(zend_object *obj) {
+	return (sysvmsg_queue_t *)((char *)(obj) - XtOffsetOf(sysvmsg_queue_t, std));
+}
+
+#define Z_SYSVMSG_QUEUE_P(zv) sysvmsg_queue_from_obj(Z_OBJ_P(zv))
+
+static zend_object *sysvmsg_queue_create_object(zend_class_entry *class_type) {
+	sysvmsg_queue_t *intern = zend_object_alloc(sizeof(sysvmsg_queue_t), class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
```

Comment:
since your declare the class as FINAL, if not property are declared, then this one should be unnecessary

Reply:
anyway, this just a hint,  leave it there is also okey in case of we introduce some properties later.

Reply:
I'm okay either way here.

Reply:
I left this as-is (because of the future-proofing purposes mentioned by Xinchen), but thanks for the hints! :)